src/clr-angular/forms/range/range-container.ts

```diff
@@ -0,0 +1,121 @@
+import { RangeControllerService } from './providers/range-controller.service';
+
+@Component({
+  selector: 'clr-range-container',
+  template: `
+        <ng-content select="label"></ng-content>
+        <label *ngIf="!label && addGrid()"></label>
+        <div class="clr-control-container" [ngClass]="controlClass()">
+            <div class="clr-range-wrapper" [class.progress-fill]="hasProgress">
+                <ng-content select="[clrRange]"></ng-content>
+                <span
+                  *ngIf="hasProgress"
+                  class="fill-input"
+                  [style.width]="getRangeProgressFillWidth()"
+                ></span>
+                <clr-icon *ngIf="invalid" class="clr-validate-icon" shape="exclamation-circle" aria-hidden="true"></clr-icon>
+            </div>
+            <ng-content select="clr-control-helper" *ngIf="!invalid"></ng-content>
+            <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
+        </div>
+    `,
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-form-control-disabled]': 'control?.disabled',
+    '[class.clr-row]': 'addGrid()',
+  },
+  providers: [IfErrorService, NgControlService, ControlIdService, ControlClassService, RangeControllerService],
+})
+export class ClrRangeContainer implements DynamicWrapper, OnDestroy {
+  private subscriptions: Subscription[] = [];
+  invalid = false;
+  _dynamic = false;
+  @ContentChild(ClrLabel, { static: false })
+  label: ClrLabel;
+  control: NgControl;
+
+  private _hasProgress: boolean = false;
+
+  @Input('clrRangeHasProgress')
+  set hasProgress(val: boolean) {
+    const valBool = !!val;
+    if (valBool !== this._hasProgress) {
+      this._hasProgress = valBool;
+    }
+  }
+
+  get hasProgress() {
+    return this._hasProgress;
+  }
+
+  constructor(
+    private ifErrorService: IfErrorService,
+    @Optional() private layoutService: LayoutService,
+    private controlClassService: ControlClassService,
+    private ngControlService: NgControlService,
+    private rangeControllerService: RangeControllerService,
+    private renderer: Renderer2
+  ) {
+    this.subscriptions.push(
+      this.ifErrorService.statusChanges.subscribe(invalid => {
+        this.invalid = invalid;
+      })
+    );
+    this.subscriptions.push(
+      this.ngControlService.controlChanges.subscribe(control => {
+        this.control = control;
+      })
+    );
+  }
+
+  getRangeProgressFillWidth(): string {
+    let inputWidth = this.getInputWidth();
+    let inputMinValue = this.rangeControllerService.rangeMinValue;
+    let inputMaxValue = this.rangeControllerService.rangeMaxValue;
+
+    let inputMiddle = (inputMinValue + inputMaxValue) / 2;
+    let inputValue = !!this.control && this.control.value !== undefined ? this.control.value : inputMiddle;
+    let valueAsPercent = (inputValue - inputMinValue) * 100 / (inputMaxValue - inputMinValue);
+
+    return valueAsPercent * inputWidth / 100 + 'px';
```

Comment:
Would using a percent work here?

Reply:
I tried to make this work with percent, but the `input` and the `span` element should be in the same container. This way, the `span` can be separate. I don't think that this affects the scalability, because `this.renderer.selectRootElement('[clrRange]').offsetWidth` return the width of the input.